src/telemetry/telemetry.cpp

```diff
@@ -41,13 +42,17 @@ std::string GetMachineId() {
 
 const int kMaxBatchSize = 100;
 
-Telemetry::Telemetry(std::string url, std::filesystem::path storage_directory,
+Telemetry::Telemetry(std::string url, std::filesystem::path storage_directory, std::filesystem::path run_id_directory,
                      std::chrono::duration<int64_t> refresh_interval, const uint64_t send_every_n)
     : url_(std::move(url)),
       uuid_(utils::GenerateUUID()),
       machine_id_(GetMachineId()),
       send_every_n_(send_every_n),
       storage_(std::move(storage_directory)) {
+  memgraph::kvstore::KVStore run_id_storage(run_id_directory);
+  run_id_storage.Put("run_id", uuid_);
+  run_id_storage.~KVStore();
```

Comment:
Aha, you used kvstore as a way to pass/communicate the `run_id`? That seems like a hack, can you pass that info in some other way because relying on disk for this type of thing doesn't make much sense to me.

Reply:
E.g. putting that into `SessionData` object seems much better 😄